lib/livebook_web/live/session_live/input_cell_settings_component.ex

```diff
@@ -0,0 +1,56 @@
+defmodule LivebookWeb.SessionLive.InputCellSettingsComponent do
+  use LivebookWeb, :live_component
+
+  alias Livebook.Session
+
+  @impl true
+  def update(assigns, socket) do
+    cell = assigns.cell
+
+    socket =
+      socket
+      |> assign(assigns)
+      |> assign(name: cell.name, type: cell.type)
```

Comment:
I am worried this can be problematic in a situation where the parent LiveView re-renders the component, updating the fields set by the user. What do you think about:

```elixir
socket
|> assign_new(:name, fn -> cell.name end)
|> assign_new(:type, fn -> cell.type end)
```

Reply:
Good point, this is a modal so it's fine if we just initialize the values once and ignore any changes, we want the user's input afterall :D